src/Handler/AliasHandler.php

```diff
@@ -39,26 +40,44 @@ public function __construct(ObjectManager $manager, AliasCreator $creator)
 
     /**
      * @param array $aliases
+     * @param int   $limit
      *
      * @return bool
      */
-    public function checkAliasLimit(array $aliases): bool
+    public function checkAliasLimit(array $aliases, int $limit = self::ALIAS_LIMIT_CUSTOM): bool
     {
-        return (count($aliases) < self::ALIAS_LIMIT) ? true : false;
+        return (count($aliases) < $limit) ? true : false;
     }
 
     /**
-     * @param User        $user
-     * @param null|string $localPart
+     * @param User $user
      *
      * @return Alias|null
      *
      * @throws ValidationException
      */
-    public function create(User $user, ?string $localPart): ?Alias
+    public function createRandom(User $user): ?Alias
     {
-        $aliases = $this->repository->findByUser($user);
-        if ($this->checkAliasLimit($aliases)) {
+        $aliases = $this->repository->findRandomByUser($user);
+        if ($this->checkAliasLimit($aliases, self::ALIAS_LIMIT_RANDOM)) {
+            return $this->creator->create($user, null);
+        }
+
+        return null;
+    }
+
+    /**
+     * @param User   $user
+     * @param string $localPart
+     *
+     * @return Alias|null
+     *
+     * @throws ValidationException
+     */
+    public function createCustom(User $user, string $localPart): ?Alias
+    {
+        $aliases = $this->repository->findCustomByUser($user);
+        if ($this->checkAliasLimit($aliases, self::ALIAS_LIMIT_CUSTOM)) {
```

Comment:
This check is not sufficient. It would allow a collision between an existing random alias and a new custom alias.